### PR TITLE
Improve retry logic

### PR DESCRIPTION
**Pull request by aarshkshah1992**
Improve retry logic

### CODE REVIEW

File: caboose.go [hunk 1]
```diff
@@ -85,11 +85,12 @@ const DefaultLoggingInterval = 5 * time.Second
 const DefaultSaturnOrchestratorRequestTimeout = 30 * time.Second
 
 const DefaultSaturnBlockRequestTimeout = 19 * time.Second
-const DefaultSaturnCarRequestTimeout = 30 * time.Minute
+
+const DefaultSaturnCarRequestTimeout = 1 * time.Minute
```
Comment (aarshkshah1992) [inline caboose.go:89]: Max P95 for a successful CAR fetch over the last 24 hours in prod is 30 seconds:

https://protocollabs.grafana.net/d/bxJz6bJ4k/bifrost-gw-metrics?orgId=1&from=now-24h&to=now&editPanel=134

File: caboose.go [hunk 2]
```diff
@@ -85,11 +85,12 @@ const DefaultLoggingInterval = 5 * time.Second
...
 const DefaultSaturnMirrorRequestTimeout = 30 * time.Second
 
 // default retries before failure unless overridden by MaxRetrievalAttempts
-const defaultMaxRetries = 3
+const defaultMaxRetries = 2
```
Comment (aarshkshah1992) [inline caboose.go:93]: ~99% of ALL requests that succeed do so in the first two retries:

https://protocollabs.grafana.net/d/6g0_YjBVk/bifrost-caboose-staging?orgId=1&from=1692841797171&to=1692863397171&viewPanel=98

File: fetcher.go [hunk 3]
```diff
@@ -236,9 +236,9 @@ func (p *pool) fetchResource(ctx context.Context, from string, resource string,
 	// which is the amount of time without any NEW data from the server, but
 	// that can be added later. We need both because a slow trickle of data
 	// could take a large amount of time.
-	requestTimeout := DefaultSaturnCarRequestTimeout
+	requestTimeout := time.Duration(DefaultSaturnCarRequestTimeout.Milliseconds()/int64(attempt+1)) * time.Millisecond
```
Comment (aarshkshah1992) [inline fetcher.go:239]: Reduces timeout duration with each retry.

File: caboose.go (hunk 1 above)
Comment (willscott) [reply caboose.go:89]: does this mean that a large movie won't work anymore?


I don't think this is an acceptable limit.

File: caboose.go (hunk 2 above)
Comment (willscott) [reply caboose.go:93]: this seems fine

File: fetcher.go (hunk 3 above)
Comment (willscott) [reply fetcher.go:239]: this seems okay

File: caboose.go (hunk 1 above)
Comment (aarshkshah1992) [reply caboose.go:89]: Hmmm, I see what you mean.

Comment (aarshkshah1992) [reply caboose.go]: @willscott Reverted this change.

**Review by willscott — APPROVED** [review]

**Pull request merged**